### PR TITLE
Add return types affine add expr

### PR DESCRIPTION
**Pull request by sunnygurm**
## Description
While getting familiar with the repository and codebase, I ran `ruff check --select ANN201 ./cvxpy` to see how many issues existed relating to return type annotations for public functions. 

I found 933 such instances (see attached [ann201_errors.txt](https://github.com/cvxpy/cvxpy/files/13667751/ann201_errors.txt)). 

More details for ANN201 errors and why it might be good to fix them can be found [here](https://docs.astral.sh/ruff/rules/missing-return-type-undocumented-public-function/). 

This PR adds return type annotations that were missing in `cvxpy/atoms/affine/add_expr.py`. 

*My questions for the team*
- Do you want to see more of these types of PRs? 
- If so, given the large number of similar errors, could we put this into a 'Good first issue' type of issue?

## Type of change
- [ ] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [ ] Bug fix
- [x] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [ ] Add our license to new files.
- [ ] Check that your code adheres to our coding style.
- [ ] Write unittests.
- [ ] Run the unittests and check that they’re passing.
- [ ] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

File: cvxpy/atoms/affine/add_expr.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def shape_from_args(self) -> Tuple[int, ...]:
         """
         return u.shape.sum_shapes([arg.shape for arg in self.args])
 
-    def expand_args(self, expr):
+    def expand_args(self, expr: Any) -> List:
```
Comment (SteveDiamond) [inline cvxpy/atoms/affine/add_expr.py:44]: I believe expr has to be a cvxpy Expression

**Comment by SteveDiamond** [comment]
Sure, this is helpful. We want all the Python code to have types. It could be a good first issue.

File: cvxpy/atoms/affine/add_expr.py [hunk 2]
```diff
@@ -41,7 +42,7 @@ def shape_from_args(self) -> Tuple[int, ...]:
         """
         return u.shape.sum_shapes([arg.shape for arg in self.args])
 
-    def expand_args(self, expr):
+    def expand_args(self, expr: Expression) -> List:
```
Comment (SteveDiamond) [inline cvxpy/atoms/affine/add_expr.py:45]: List[Expression]

File: cvxpy/atoms/affine/add_expr.py (hunk 1 above)
Comment (sunnygurm) [reply cvxpy/atoms/affine/add_expr.py]: Updated

File: cvxpy/atoms/affine/add_expr.py (hunk 2 above)
Comment (SteveDiamond) [reply cvxpy/atoms/affine/add_expr.py]: Sorry not quite what I meant

Comment (sunnygurm) [reply cvxpy/atoms/affine/add_expr.py]: Oh! My fault, yeah that makes sense now that I look at the function return.

**Review by SteveDiamond — APPROVED** [review]

**Pull request merged**